### PR TITLE
Download log

### PR DESCRIPTION
**Pull request by mdellweg**
Attempting to supersede https://github.com/pulp/pulp_ansible/pull/945

this is for https://issues.redhat.com/browse/AAH-1558

AAH-1558

### CODE REVIEW

File: pulp_ansible/app/models.py [hunk 1]
```diff
@@ -211,6 +222,28 @@ class Meta:
...
+    user = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE,
+        null=True,
+        related_name="download_logs",
+    )
+    ip = models.GenericIPAddressField()
+    extra_data = models.JSONField(null=True)
+    user_agent = models.TextField()
+    repository = models.ForeignKey(
```
Comment (newswangerd) [inline pulp_ansible/app/models.py:242]: Would repository version be more useful?

**Review by newswangerd — COMMENTED** [review]
This looks good to me.

Now that I have a better understanding of repository versions, I think storing the repo version in the download log would be more useful than the repository.

File: pulp_ansible/app/models.py (hunk 1 above)
Comment (mdellweg) [reply pulp_ansible/app/models.py:242]: Repository versions might be a bit more volatile if you are using `retain_repo_versions`. But if you like, you can get both.

**Comment by newswangerd** [comment]
@mdellweg I'm running into circular dependecy issues. I think Levente got around it with this: https://github.com/pulp/pulp_ansible/pull/945/files#diff-ede042cbf5a92f6001b82d26b276d87a1f7fe3fa28e8a27913b219a83880e2a2R223

**Review by newswangerd — APPROVED** [review]
Everything seems to be working. Just need to apply the changes I mentioned above to resolve the circular import.

**Pull request merged**